server/plugin/hooks.go

```diff
@@ -70,6 +70,11 @@ func (p *Plugin) MessageHasBeenPosted(c *plugin.Context, post *model.Post) {
 
 	token, err := p.ParseAuthToken(user.OAuth2Token)
 	if err != nil {
+		if err.Error() == ErrorMessageAuthenticationFailed {
+			p.API.LogError(err.Error())
+			p.Ephemeral(mattermostUserID, post.ChannelId, ReconnectMessage)
```

Comment:
check if we can use `p.logAndSendErrorToUser` here like used below

Reply:
I think we can't do this because in `p.logAndSendErrorToUser` they are always using `GenericErrorMessage` while sending an ephemeral message to a user.

Reply:
Discussed